### ISSUE TITLE
rename `isa` to `instanceof`?

### ISSUE THREAD

**Issue by StefanKarpinski** [JuliaLang/julia#19524]
It should be more familiar to Python programmers and potentially less confusing?

**Comment by quinnj** [comment]
Java also has `instanceof`, though it it used infix.

**Comment by KristofferC** [comment]
Instance sounds more like it is related to classes to me. Personally I like `isa`.

**Comment by yuyichao** [comment]
Would be `isinstance` for python. What's the confusion about the current one? (especially since `is` is deprecated now)

**Comment by StefanKarpinski** [comment]
Just witnessed a room full of confusion about what `isa` meant, so I thought I'd throw it out there.

**Comment by ararslan** [comment]
Do you know what about it they found confusing? It's always seemed quite clear to me.

**Comment by johnmyleswhite** [comment]
I think people often confuse `isa` with `<:`. I've done it myself a bunch of times.

**Comment by StefanKarpinski** [comment]
Deprecation of `is` definitely helps. Maybe it's just fine; people do seem to confuse it with `<:`. I'm not sure what the solution is. I think that `::` and `<:` are fairly clear (although sometimes those get confused too). Maybe there's nothing to be done here.

**Comment by yuyichao** [comment]
What I do want from time to time is an operator for `isa`.

**Comment by JeffBezanson** [comment]
I find `instanceof` a bit too long.

**Comment by pabloferz** [comment]
If `isa` were also an operator (as `in` is, for example), that probably would help to avoid for some people to get confused, with the benefit of being easy to type. E.g.

```julia
a = 1
if a isa Int
    #do something
end
```

seems pretty clear to me.

**Comment by simonbyrne** [comment]
I like the fact that most predicates begin with `is`.

**Comment by JeffreySarnoff** [comment]
In another context, I used  _is_ followed by _a kind of_ in anagram form:
```julia
fourtytwo = "42"
isako( fourtytwo, Integer )
```

**Comment by oxinabox** [comment]
I can never remember the name of the `isa` function.

I then to write: `foo |> typeof <: Integer`
which is I guess pseudo-infix.

**Comment by iamed2** [comment]
> Just witnessed a room full of confusion about what isa meant, so I thought I'd throw it out there.
Was the confusion about the behaviour or just how to read the function name? I wouldn't mind a change to `is_a` (a few people at Invenia hate the "underscore case except sometimes not" rule).

**Comment by StefanKarpinski** [comment]
I dunno. Probably not worth changing. I agree that the underscore rule can be annoying. @JeffBezanson hates underscores :P. I would be ok with using underscores and then just trying to delete all names that have them in Base over time.

**Comment by ararslan** [comment]
I've implemented the infix operator version of `isa` in #19677, as discussed here by Yichao and Pablo, if anyone's interested in taking a look.
